internal/xds/translator/cluster.go

```diff
@@ -191,8 +209,13 @@ func buildXdsCluster(args *xdsClusterArgs) (*buildClusterResult, error) {
 		cluster.TransportSocket = args.tSocket
 	}
 
+	requiresAutoHTTPConfig := false
 	for i, ds := range args.settings {
 		if ds.TLS != nil {
+			if !requiresAutoHTTPConfig && len(ds.TLS.ALPNProtocols) == 0 {
```

Comment:
this means, if ALPN is set to h2, it won't actually connect over h2, is that right ?

Reply:
if alpnProtocols = [h2], requiresAutoHTTPConfig is false, it should be same as the old logic which means use h2. 
what I missed?

https://github.com/envoyproxy/gateway/blob/5ebc00489ff2284a947b83a4fec9e3594098d79f/internal/gatewayapi/testdata/clienttrafficpolicy-for-tcp-listeners.out.yaml#L253-L254

Reply:
imo we should use this for the default case when tls is non nil